src/parser.js

```diff
@@ -34,6 +33,36 @@ function getSource (startLine, startColumn, endLine, endColumn) {
     };
 }
 
+function ensureObject (obj, ...props) {
+    const prop = props[0];
+
+    if (!obj[prop]) {
+        obj[prop] = {};
+    }
+
+    if (props.length > 1) {
+        props.splice(0, 1);
+
+        ensureObject(obj[prop], ...props);
```

Comment:
I'd prefer a non-recursive definition for these, a while loop is cheaper, the code isn't that complicated, and these functions will get called a lot.